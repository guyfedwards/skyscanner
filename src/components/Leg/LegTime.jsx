import React from 'react';
import BpkText from 'bpk-component-text';
import styled from 'styled-components';
import propTypes from 'prop-types';

const Root = styled.div`
  display: flex;
  flex-direction: column;
  width: 40px;

  margin-right: 15px;
  margin-left: 15px;
`;

const Airport = styled(BpkText)`
  color: #aba9a9;
`;

const LegTime = ({ time, airport }) => {
  const localTime = new Date(time).toLocaleTimeString('en-GB', {
    hour: '2-digit',
    minute: '2-digit',
  });

  return (
    <Root>
      <BpkText textStyle="lg">{localTime}</BpkText>
      <Airport textStyle="lg">{airport}</Airport>
    </Root>
  );
};

LegTime.propTypes = {
  airport: propTypes.string.isRequired,
  time: propTypes.string.isRequired,
};

export default LegTime;
