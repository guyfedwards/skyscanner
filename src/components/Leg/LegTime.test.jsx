import React from 'react';
import ReactDOM from 'react-dom';
import ShallowRenderer from 'react-test-renderer/shallow';

import LegTime from './LegTime';

describe('LegTime', () => {
  it('should render without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(<LegTime time="2020-10-31T17:00" airport="LTN" />, div);
  });

  it('should render correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<LegTime time="2020-10-31T17:00" airport="LTN" />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
