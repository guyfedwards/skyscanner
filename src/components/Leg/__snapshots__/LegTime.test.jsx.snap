// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LegTime should render correctly 1`] = `
<ForwardRef(styled.div)>
  <BpkText
    bold={false}
    className={null}
    tagName="span"
    textStyle="lg"
  >
    05:00 PM
  </BpkText>
  <ForwardRef(Styled(BpkText))
    textStyle="lg"
  >
    LTN
  </ForwardRef(Styled(BpkText))>
</ForwardRef(styled.div)>
`;
